doozerlib/cli/release_gen_payload.py

```diff
@@ -504,6 +506,16 @@ def update_single_arch_istags(apiobj: oc.APIObject):
                     if adding_tags:
                         logger.warning(f'The following tag names are net new to {imagestream_namespace}:{imagestream_name}: {adding_tags}')
 
+            if publish:
```

Comment:
If `--apply` is not used this will cause an error because the referenced image name doesn't exist.